### PR TITLE
aws.ecs-task-definition | config support

### LINKED ISSUE #6506: Add config support for ECS
**Is your feature request related to a problem? Please describe.**
We have a lot of ecs task definitions(100K+) in our environment and are running into api throttling issue for controls around task definitions.

**Describe the solution you'd like**
AWS recently rolled out Config support for ECS resources(https://aws.amazon.com/about-aws/whats-new/2021/02/aws-config-supports-amazon-container-services/)

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by PratMis**
Closes #6506 

note: Response received from AWS config doesn't seem consistent with the boto3 response. Hence, I had to override the response to add `taskDefinitionArn` key that's used in all ecs-task-definition actions(tag/remove-tag/delete).

### CODE REVIEW

File: c7n/resources/ecs.py [hunk 1]
```diff
@@ -513,6 +506,31 @@ def augment(self, resources):
         return results
 
 
+class ConfigECSTaskDefinition(ConfigSource):
+
+    def load_resource(self, item):
+        resource = super().load_resource(item)
+        if 'TaskDefinitionArn' in resource.keys():
+            resource['taskDefinitionArn'] = resource['TaskDefinitionArn']
```
Comment (kapilt) [inline c7n/resources/ecs.py:514]: afaics we should do this unconditionally as well as remove the previous key

File: c7n/resources/ecs.py [hunk 2]
```diff
@@ -513,6 +506,31 @@ def augment(self, resources):
         return results
 
 
+class ConfigECSTaskDefinition(ConfigSource):
+
+    def load_resource(self, item):
+        resource = super().load_resource(item)
```
Comment (kapilt) [inline c7n/resources/ecs.py:512]: the keys on all of the attribute are off. we probably just want to just unconditionally remap them for top level elements and containers array.

Comment (kapilt) [reply c7n/resources/ecs.py:512]: more specifically config returns ContainerAttributes and we want containerAttributes to match describe output.

Comment (PratMis) [reply c7n/resources/ecs.py:512]: thanks for the feedback @kapilt. Yeah, that was something I was thinking about but wasn't sure. I'll make the changes shortly

**Comment by PratMis** [comment]
@kapilt looks ECS hasn't made it here yet: https://github.com/boto/botocore/blob/a8d13bb5b70283f69dd518f277b050a86a40a794/botocore/data/config/2014-11-12/service-2.json#L6238 and its causing the test to fail

File: c7n/resources/ecs.py (hunk 2 above)
Comment (PratMis) [reply c7n/resources/ecs.py:512]: Remaped top level keys

**Comment by kapilt** [comment]
> @kapilt looks ECS hasn't made it here yet: https://github.com/boto/botocore/blob/a8d13bb5b70283f69dd518f277b050a86a40a794/botocore/data/config/2014-11-12/service-2.json#L6238 and its causing the test to fail

you should be able to file a support request with the service team via your tam on that. the service team controls those files. its also possible that maybe config support for ecs isn't fully baked if its not in that resource type list which is needed for some api interactions as its used to validate parameters.

**Comment by kapilt** [comment]
alternatively we maintain a white list of various resources that config supports but hasn't bothered to update its sdk correctly for that we explicitly check in this test.

**Comment by PratMis** [comment]
> > @kapilt looks ECS hasn't made it here yet: https://github.com/boto/botocore/blob/a8d13bb5b70283f69dd518f277b050a86a40a794/botocore/data/config/2014-11-12/service-2.json#L6238 and its causing the test to fail
> 
> you should be able to file a support request with the service team via your tam on that. the service team controls those files. its also possible that maybe config support for ecs isn't fully baked if its not in that resource type list which is needed for some api interactions as its used to validate parameters.

Thanks @kapilt! Trying to bring this up to our TAM. Will keep you posted.

**Comment by PratMis** [comment]
linking botocore issue: https://github.com/boto/botocore/issues/2323

File: c7n/resources/ecs.py (hunk 2 above)
Comment (kapilt) [reply c7n/resources/ecs.py:512]: it should be a recursive helper function else it wont get to container definitions which aren't top level keys as their in the containers array.

Comment (PratMis) [reply c7n/resources/ecs.py:512]: Sounds good!

**Comment by kapilt** [comment]
i've updated the pr to correctly normalize and to work around the config service sdk issue.

File: c7n/resources/ecs.py (hunk 2 above)
Comment (kapilt) [reply c7n/resources/ecs.py]: i went ahead and added this

**Comment by PratMis** [comment]
oops, sorry i didn't get back to this thinking AWS would have to fix boto before we could merge this PR. Thanks @kapilt 👍

**Pull request merged**